document/js-api/index.bs

```diff
@@ -687,6 +695,62 @@ which can be simultaneously referenced by multiple {{Instance}} objects. Each
     1. Return undefined.
 </div>
 
+<h3 id="globals">Globals</h3>
+
+<pre class="idl">
+dictionary GlobalDescriptor {
+  required USVString type;
+  required Number value;
+  boolean mutable;
+};
+
+[LegacyNamespace=WebAssembly, Constructor(GlobalDescriptor descriptor), Exposed=(Window,Worker,Worklet)]
+interface Global {
+  // TODO: How to define @@toPrimitive hook?
```

Comment:
I think you'll have to define this outside of the IDL, in some unconnected text which comes later.

Do you actually want to hook on `@@toPrimitive`? A simple `valueOf` hook might be enough. The JS ToPrimitive operation calls `@@toPrimitive` if it exists, but falls back to `valueOf` later. `valueOf` was made to cast to Number, and `@@toPrimitive` was added as a more advanced form to match `Date`'s unusual casting semantics.

Reply:
What would be the impact on this as we add more types to the language and allow more values to be extracted here?  Looking beyond i64, suppose we add globals that can contain reference types later (a la host bindings / gc proposals).  Will valueOf continue to serve us?

Reply:
I think that @littledan is right, `valueOf` should be enough. Neither `@@toPrimitive` nor `valueOf` will make much sense for globals over reference types, because JS expects them to return primitive values. But it is really just a backwards compatibility hack anyway. In general, people should be using the explicit getter.

Reply:
OK, I've added a `valueOf` method, PTAL.

Reply:
@lars-t-hansen For something that converts to BigInt, `valueOf` is still appropriate. For something which is naturally converted to a String, make a `toString` method and omit a `valueOf` method. If more JS primitive types are added, we may need to hook into things in a different way.